### DISCUSSION TITLE
Is it possible to Wallbash telegram-desktop

### DISCUSSION THREAD

**Discussion by KitsuneKode** [prasanthrangan/hyprdots#1490] [Q&A]
Does the dotfile automatically wallbash the telegram? I am not sure if it does please show me how, I tried a few things but it didn't do the trick. I may be wrong but is it like telegram does not supports wallbash?

**Comment by kRHYME7** [comment]
No one made a template for telegram.Do telegram have customization capabilities out of the box?.

**Reply by KitsuneKode** [reply]
Not sure about that! I recently tried [64Gram](https://github.com/TDesktop-x64/tdesktop). It has some more enhancement than that of the official release, but as far as i am aware only has some default theme support. 


But I came across two other forks of the official telegram which may have the customizable capacities
[Materialgram](https://github.com/kukuruzka165/materialgram) 
[Kotatogram](https://github.com/kotatogram/kotatogram-desktop)

**Reply by kRHYME7** [reply]
Is telegram QT app? if it is it should follow like how QT apps are rendered. How did you install it? flatpak? aur?

**Reply by KitsuneKode** [reply]
did install from flatpak!

But I think telegram is a qt app!

**Reply by kRHYME7** [reply]
ops My bad, it's Qt but it is not respecting kvantum colors. @KitsuneKode  It would be helpful if we can find a theme-able client for telegram. I'll try to check the ones you provide.

**Reply by KitsuneKode** [reply]
that would be great! Will search for a better fork as well, will let you know if a better fork exists than the ones given above!

**Reply by kRHYME7** [reply]
So I tested those and unfortunately none of them is customizable. Don't hesitate to ping me in case.

**Reply by KitsuneKode** [reply]
Sure! I will leave this discussion open just in case someone has any update on this!

**Reply by Enigma1309** [reply]
telegram official is customizable but idk the way to automatically script the process
https://github.com/catppuccin/telegram

**Reply by kRHYME7** [reply]
Cool 

https://github.com/catppuccin/telegram?tab=readme-ov-file#-development-for-the-desktop-version

>  Rebuild flavors after editing the template file.

I do not know if this is dynamic

**Reply by AndrejGorodnij** [reply]
> telegram official is customizable but idk the way to automatically script the process https://github.com/catppuccin/telegram

There is few projects - `walogram` and `wal-telegram-rs`, that generates themes for Telegram desktop based on Wal colors... I start using @kRHYME7 dots yesterday, and was can't figure out why walogram doesn't works, until I've realized that this dots use `swwwallbash` instead of original `wal`... so walogram can't get colors from default wal directory...

Will be great if swwwalbash and wallbash will overwrite default .cashe/wal/* files for maximal compatibility, btw ^_^

What about dynamic - Telegram Desktop changing colors on every theme file rewrite, so we can use it as "raw" method, but I think official theme template has some options (I'm not sure) for native dynamic changing between Light/Dark/

**Reply by kRHYME7** [reply]
My bad for not updating this yet but here you go. I actually managed to hydrate it lol.

https://github.com/hyde-project/telegram

still a WIP though so please keep in touch with the repo for now 


Lemme correct my statement too. Telegram is very customizable.
